src/lib.rs

```diff
@@ -32,9 +32,42 @@
 mod windows;
 
 use std::env;
+use std::ffi::OsString;
 use std::io;
 use std::path::{Path, PathBuf};
 
+/// Permits parameterising the home functions via the _from variants - used for
+/// in-process unit testing by rustup.
+pub trait Env {
+    fn home_dir(&self) -> Option<PathBuf>;
```

Comment:
Nitpick: It would be nice to document those trait methods.
Requirements? Conditions? Panics? Or just point the doc to its defer method? In this case `std::env::home_dir`.

Reply:
Sure, I can do that. It isn't std::env::home_dir though, it is home_inner; which on Un*x family OS's is std::env::home_dir, but not on Windows.